bin/varnishd/cache/cache_tcp_pool.c

```diff
@@ -571,6 +573,20 @@ tmo2msec(vtim_dur tmo)
 	return ((int)floor(tmo * 1000.0));
 }
 
+static int
+vtp_preamble(const struct tcp_pool *tp, int s)
+{
+	int r;
+
+	if (s < 0 || tp->preamble == NULL)
+		return (s);
+	r = write(s, tp->preamble, tp->preamble_l);
```

Comment:
as discussed:
Yes, I want to have a look how passing the IOV up to V1L could look like and will update the patch with a suggestion wrt this API

Reply:
FTR: The preamble will _very_ likely be local / close by because the preamble will tell someone making the connection on our behalf where to actually connect to. So the additional `write()` really should not matter at all.

Yet I still want to explore better integration with V1L

Reply:
I did have a look and think that keeping the extra write is the better option for these reasons:
* the `write()` happens only for connection setup where cost is already high
* the primary use case for preambles are via backends using UDS
* V1L is http/1 specific and VTP should not interfere with the protocol

Reply:
Can this `write()` block forever, or does it respect `connect_timeout`? Does it account for time already spent establishing the connection?

Reply:
this works analogous to the existing proxy header send code: For a non-zero timeout, `VTCP_Connect()` has put the socket into non blocking mode and polled it for write. Yet I think you have a point for the case that we send a preamble _and_ a proxy header. Will look again later